### ISSUE TITLE
Roadmap to 1.0.0

### ISSUE THREAD

**Issue by mAAdhaTTah** [valtech-nyc/brookjs#82]
* [x] 0.8.x - bug fixes
* [x] 0.9.0 - Remove current animation API, migrate `ofType` -> `thru` w/ function to be named later
* [x] 0.10.0 - Release animation API & brookjs-desalinate (testing tools)
* [x] 0.11.0 - Release brookjs-cli. Break up brookjs into more smaller packages
* [ ] 0.12.0 - Release silt & build tools (loader, bsfy transform, etc.), initial public implementation of VDOM-based rendering
    * We need to switch from string-based to VDOM-based rendering. I'm considering a JSX-based approach here, or we can implement a handlebars parser for the current templating, but we only have time to go with one. I'll be drawing up an RFC for the JSX-based approach before continuing forward.
* [ ] 0.13.0 - Deprecate string-based templates
* [ ] 0.14.0 - Remove support for string based templates (assuming silt is stable enough to replace it)

Testing tools will also be released when they're ready. We're still trying to decide on which testing framework to recommend and use.

**Comment by mAAdhaTTah** [comment]
We're going to bump silt to 0.11.0. The CLI will also go out in 0.9.0, but it may be feature incomplete at that point.

**Comment by mAAdhaTTah** [comment]
We'll be floating a strawman proposal for a JSX-based API, which we may transition to from the current handlebars-based API. We'll need to ensure there's a proper transition path, but it would mean the non-node SSR approach would no longer be possible.

**Comment by mAAdhaTTah** [comment]
CLI is now out as version 0.11.0. That's useful and feature complete as-is but can be fleshed out over time. We'll be drawing up a proposal for a JSX-based API.

0.11.0 could be considered stable and feature-complete, with bug fixes for any issues it may have with the current handlebars-based template implementation. The next step is to transition from string-based to VDOM-based rendering. We've started that with silt, but I'm not entirely convinced a handlebars-based implementation is best, so I've paused that development in order to float an alternative JSX-based proposal. This would ease the learning process from React to `brookjs`, since it's so heavily inspired by React in the first place, as well as simplify a couple pieces of complexity currently involved in matching the handlebars template to the component definition.
